### ISSUE TITLE
Performance, UI and accessibility issues 

### ISSUE THREAD

**Issue by abhishekakade** [zero-to-mastery/ZtM-Job-Board#986]
- Off screen images need to be deferred and lazy loaded. This is affecting the performance a lot, especially on slower networks. 
- Unnecessary fetching and rendering of users & images on window resize. UseEffect can be replaced by CSS.
- Render blocking Google fonts and leaflet.css. 
- Long overflowing text goes out of card containers. 
- Searchbox does not have label for screen readers.
